### PR TITLE
Wrf config to json

### PR DESCRIPTION
**Pull request by crdanielbusch**
## Description

 At the moment the CMAQ config files are json and the WRF config files are json files with comments. The handling of the configuration file should be unified for WRF and CMAQ.

- [x] Remove comments from WRF config files -> the files are now valid json files. All comments are in the config class.
- [x] Remove the step that would read the config as a string
- [x] Change config access in `setup_for_wrf.py` from config `dict` to `WRFconfig` object

## Checklist

Please confirm that this pull request has done the following:

- [x] Tests added
- [ ] Documentation added (where applicable)

### CODE REVIEW

File: src/setup_runs/config_read_functions.py [hunk 1]
```diff
@@ -37,58 +37,34 @@ def boolean_converter(
...
-    try:
-        # strip out the comments
-        input_str = re.sub(r"#.*\n", "\n", input_str)
-        return json.loads(input_str)
-    except Exception as e:
-        print("Problem parsing in configuration file")
-        print(e)
-        sys.exit()
+#
+# def read_config_file(configFile: str) -> str:
```
Comment (lewisjared) [inline src/setup_runs/config_read_functions.py:41]: Is this still needed? Otherwise it can be deleted

File: scripts/setup_for_wrf.py [hunk 2]
```diff
@@ -31,98 +29,75 @@
...
+def decode_bytes(x) :
+    if isinstance(x, bytes) :
         x = x.decode("utf-8")
     return x
 
 
-def purge(dir, pattern):
-    for f in os.listdir(dir):
-        if re.search(pattern, f) is not None:
+def purge(dir, pattern) :
```
Comment (lewisjared) [inline scripts/setup_for_wrf.py:55]: Can you ruff this file. It looks like there has been some weird non-ruff formatting applied

File: src/setup_runs/config_read_functions.py (hunk 1 above)
Comment (crdanielbusch) [reply src/setup_runs/config_read_functions.py]: It's deleted.

File: scripts/setup_for_wrf.py (hunk 2 above)
Comment (crdanielbusch) [reply scripts/setup_for_wrf.py]: That's the formatting from the "Reformat code" button in PyCharm that should be based on ruff. But it's obviously different, so maybe something is wrong with the plugin. It's probably better to just use the make command

**Comment by crdanielbusch** [comment]
@lewisjared was there anything else I should change? If not, can you click on approve so I can merge.

File: scripts/setup_for_wrf.py (hunk 2 above)
Comment (lewisjared) [reply scripts/setup_for_wrf.py]: Have you configured the `ruff` pycharm plugin as well?

File: tests/unit/test_read_config.py [hunk 3]
```diff
@@ -238,12 +181,68 @@ def test_009_config_object(input_str, config_path):
...
+    config["start_date"] = process_date_string(config["start_date"])
+
+    # load config object (performs all of the above steps)
+    wrf_config = load_wrf_config(config_path_wrf_docker)
+
+    assert config == asdict(wrf_config)
+
+
+def test_011_valid_CMAQ_NCI_config_file(config_path_cmaq_nci):
+    try:
```
Comment (lewisjared) [inline tests/unit/test_read_config.py:231]: These try-except blocks aren't needed

**Review by lewisjared — APPROVED** [review]

File: tests/unit/test_read_config.py (hunk 3 above)
Comment (lewisjared) [reply tests/unit/test_read_config.py:231]: Looking at your git history this should be resolved once you merge #28

**Pull request merged**